pypesto/objective/base.py

```diff
@@ -527,7 +527,9 @@ def check_grad(
             data = {**prefix_data, **data, **postfix_data}
 
         # create dataframe
-        result = pd.DataFrame(data=data)
+        result = pd.DataFrame(data=data,
+                              index=[self.x_names[ix] for ix in x_indices]
+                              if self.x_names is not None else None)
```

Comment:
```suggestion
                              index=[
                                  self.x_names[ix]
                                      if self.x_names is not None
                                      else f'x_{ix}'
                                  for ix in x_indices
                              ])
```

re: Yannik's suggestion

Reply:
Yes that could be done, but it doesn't really add any additional information beyond the default pandas range index, right?

Reply:
Yup not much more info, but e.g. if `x_indices = [5, 10, 15]` then the index will be `['x_5', 'x_10', 'x_15']` instead of `[0,1,2]`

Reply:
Oooh thats a good point!